### PR TITLE
Fixed bugs 1826, 1528, 1467, 1447, 1438, 1413: "Expected value not to be null"

### LINKED ISSUE #1826: Error when adding new block 103268 from l2sPq2O: Error: Expected value not to be null
### Description

I start ironfish node server and mining 
Ironfish start
Ironfish miners:start
20min later, the following error code appears:

image
graffiti: ironfishup

### PR DESCRIPTION
**Pull request by hamza-siddiq**
## Summary
Closes #1826, #1528, #1467, #1447, #1438, #1413 

## Testing Plan
Unit Tested, and then ran the network for a few hours. Tested successfully

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.
No
```
[ ] Yes
```
graffiti: ironfishup

### CODE REVIEW

**Comment by rohanjadvani** [comment]
How does this change fix the null exceptions in the linked tickets? Is this check related to those errors?

File: ironfish/src/syncer.ts [hunk 1]
```diff
@@ -442,7 +442,9 @@ export class Syncer {
 
       if (!this.loader) {
         this.logger.info(`Syncing orphan chain from ${peer.displayName}`)
-        this.startSync(peer)
+        if (peer.sequence !== null) {
```
Comment (NullSoldier) [inline ironfish/src/syncer.ts:445]: We should always have a peer sequence if the peer is identified. This should never happen right? Are we trying to sync from peers that are not identified?

**Comment by hamza-siddiq** [comment]
@

> How does this change fix the null exceptions in the linked tickets? Is this check related to those errors?
It checks to see if the peer sequence is not null, and then proceeds. This fixes the above bugs.

File: ironfish/src/syncer.ts (hunk 1 above)
Comment (hamza-siddiq) [reply ironfish/src/syncer.ts:445]: Yes, you should always have a peer sequence if the peer is identified. This should never happen. You are also trying to sync from peers that are not identified. Other than fixing the bug like this, there is no other method. Or to fix this bug, you’ll have to to remake hash algorithm.

**Comment by danield9tqh** [comment]
I am going to close this PR since it does not actually fix any of the issues mentioned

This does not fix https://github.com/iron-fish/ironfish/issues/1528 or https://github.com/iron-fish/ironfish/issues/1467 because those bugs are caused by the chain head being null, not the peer sequence.

It is completely unrelated to https://github.com/iron-fish/ironfish/issues/1447 and https://github.com/iron-fish/ironfish/issues/1438 because those happen when running a mining pool NOT a node.

https://github.com/iron-fish/ironfish/issues/1413 and https://github.com/iron-fish/ironfish/issues/1826 are the only two that are possibly related but this does not fix those either. 
1. These issues could be caused by the peer sequence being null, but they could also be caused by `peer.work` or `chain.head` being null as well, it is unclear
2. Even if they are caused by the peer's sequence being null, this will not fix the problem. It will only cover up the error and not surface it to the user. I think we would rather have the error surface to the user than cover it up.

**Pull request closed without merging**